### PR TITLE
Loosen and document dependency restrictions in gemspec 

### PR DESCRIPTION
**Pull request by mjgiarlo**
Also, bump rubocop-rspec up to next minor version and deal with new violations.

@samvera/hyrax-code-reviewers @jrochkind

### CODE REVIEW

File: hyrax.gemspec [hunk 1]
```diff
@@ -51,19 +51,19 @@ EOF
   spec.add_dependency 'legato', '~> 0.3'
   spec.add_dependency 'posix-spawn'
   spec.add_dependency 'jquery-ui-rails', '~> 5.0'
-  spec.add_dependency 'redis-namespace', '~> 1.5.2'
-  spec.add_dependency 'flot-rails', '~> 0.0.6'
+  spec.add_dependency 'redis-namespace', '~> 1.5'
+  spec.add_dependency 'flot-rails', '~> 0.0'
```
Comment (jcoyne) [inline hyrax.gemspec]: I'd say with a 0.0.x gem, it's very important to pin to the minor version (as it's not stable until 1.0 per semver)

File: hyrax.gemspec [hunk 2]
```diff
@@ -51,19 +51,19 @@ EOF
...
-  spec.add_dependency 'jquery-datatables-rails', '~> 3.4.0'
+  spec.add_dependency 'jquery-datatables-rails', '~> 3.4'
   spec.add_dependency 'rdf-rdfxml' # controlled vocabulary importer
   spec.add_dependency 'clipboard-rails', '~> 1.5'
   spec.add_dependency 'rails_autolink', '~> 1.1'
   spec.add_dependency 'active_fedora-noid', '~> 2.0', '>= 2.0.2'
   spec.add_dependency 'awesome_nested_set', '~> 3.1'
   spec.add_dependency 'breadcrumbs_on_rails', '~> 3.0'
-  spec.add_dependency 'kaminari_route_prefix', '~> 0.1.1'
+  spec.add_dependency 'kaminari_route_prefix', '~> 0.1'
```
Comment (jcoyne) [inline hyrax.gemspec]: same here. Please keep the pin as it was before, lest I release a breaking change 😆

File: spec/tasks/rake_spec.rb [hunk 3]
```diff
@@ -11,15 +11,15 @@
 
     it "creates a file" do
       run_task "hyrax:user:list_emails"
-      expect(File.exist?("user_emails.txt")).to be_truthy
+      expect(File).to be_exist("user_emails.txt")
```
Comment (jcoyne) [inline spec/tasks/rake_spec.rb]: "to be exist" is more awkward than what was there before. :(

Comment (mjgiarlo) [reply spec/tasks/rake_spec.rb]: Rubocop did that, and I don't like it either. Gross!

File: hyrax.gemspec (hunk 2 above)
Comment (mjgiarlo) [reply hyrax.gemspec]: Will do. Keeping my eye on you.

**Comment by jrochkind** [comment]
@mjgiarlo looks good to me if it doesn’t break anything! Should I ‘approve’ it? I haven’t tested it or anything, just looking at the PR it did what I expect, is that grounds to approve?

 You also changed some other unrelated things in spec files, don’t know if you meant those to be in the same PR.

**Pull request merged**